File: src/main/frontend/src/service/UserService.js
import axios from 'axios'; //1.

let BOARD_API_BASE_URL = "/api/user";

class UserService{
  logIn(user){
    return axios.post(BOARD_API_BASE_URL+"/logIn", user);
  }
  register(user){
    return axios.post(BOARD_API_BASE_URL+"/register", user);
  }
  confirm(user){
    return axios.post(BOARD_API_BASE_URL+"/confirm", user);
  }
  update(user){
    return axios.post(BOARD_API_BASE_URL+"/update", user);
  }
  delete(user){
    return axios.post(BOARD_API_BASE_URL+"/delete", user);
  }
  selectOne(user){
    return axios.post(BOARD_API_BASE_URL+"/selectOne", user);
  }
}
export default new UserService();

File: src/main/java/com/bitcamp/reactboard/service/UserService.java
package com.bitcamp.reactboard.service;

import com.bitcamp.reactboard.model.UserDTO;
import org.apache.ibatis.session.SqlSession;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Repository;
import org.springframework.stereotype.Service;

@Service
@Repository
public class UserService {
    private final String NAMESPACE = "com.bitcamp.reactboard.mapper.UserMapper";
    @Autowired
    private SqlSession sqlSession;

    public UserDTO auth(UserDTO u){
        return sqlSession.selectOne(NAMESPACE+".auth", u);
    }
    public int register(UserDTO u){
        return sqlSession.insert(NAMESPACE+".insert", u);
    }
    public UserDTO selectOne(int id) {
        return sqlSession.selectOne(NAMESPACE+".selectOne", id);
    }
    public int update(UserDTO u){
        return sqlSession.update(NAMESPACE+".update", u);
    }
    public void delete(int id){
        sqlSession.delete(NAMESPACE+".delete", id);
    }
}
